ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -188,6 +181,18 @@ private void initReplDump(ASTNode ast) throws HiveException {
       // move to the next root node
       currNode++;
     }
+
+    for (String dbName : Utils.matchesDb(db, dbNameOrPattern)) {
+      Database database = db.getDatabase(dbName);
+      if (database != null) {
+        if (!ReplChangeManager.isSourceOfReplication(database) && !isMetaDataOnly) {
```

Comment:
Shall check !isMetaDataOnly first to avoid extra computation in false case.

Reply:
done